ql/src/java/org/apache/hadoop/hive/ql/plan/DeferredWorkHelper.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.plan;
+
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+
+public interface DeferredWorkHelper {
+  void setupWork(CopyWork copyWork) throws HiveException;
+  void setupWork(ReplCopyWork replCopyWork) throws HiveException;
+  void setupWork(MoveWork moveWork) throws HiveException;
```

Comment:
If you have interface or abstract that is common to CopyWork, ReplCopyWork, and Move work. And a setupWork() in each of those works.
This can use be just be one setupWork(DeferredWorkHelper helper).

Interface BaseWork, CopyWork, ReplCopyWork and Move work all implement.
BaseWork.setupWork(DeferredWorkHelper helper) is the method that they all have to implement.

For instance CopyWork.setupWork(DeferredWorkHelper helper) {
     ImportSemanticAnalyzer.evaluate(helper.getReplace(), helper.getTgtPath(), helper.getWriteId(), helper.getStmtId(), helper.getHive(), helper.getCtx(), helper.getTblDesc(), helper.getInReplScope(), helper);
    this.setToPath(new Path[] { destPath });
  }
}

Reply:
Currently, the copyWork, ReplCopyWork and MoveWork do not share a common base class/interface. Do you suggest that we define a new base class/interface and then extend these classes? That base class will only have a single function that will need to be implemented.

Reply:
Yes a new interface or abstract , maybe call it BaseWork.